lib/generators.nix

```diff
@@ -175,4 +175,52 @@ rec {
                          else "<λ:{${showFnas}}>"
     else abort "toPretty: should never happen (v = ${v})";
 
+  # PLIST handling
+
+  toPLIST = x: ''
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+  '' + pprExpr "" x
+     + "\n</plist>";
+
+  pprExpr = ind: x: with builtins;
+    if isNull x then "" else
+    if isBool x then pprBool ind x else
+    if isInt x then pprInt ind x else
+    if isString x then pprStr ind x else
+    if isList x then pprList ind x else
+    if isAttrs x then pprAttrs ind x else
+    throw "invalid plist type";
```

Comment:
The error message should mirror the other pattern matching error messages in this module, namely:

```
    else abort "toPretty: should never happen (v = ${v})";
```

and 

```
    let err = t: v: abort
          ("generators.mkValueStringDefault: " +
           "${t} not supported: ${toPretty {} v}");
```

And now that I see them, they can you edit factor out the error function from the second and make all say:

```
"generators.${name}: this value is not supported: ${toPretty {} v}"
```

? :)